### PR TITLE
UseStringIsEmptyRule and unit tests

### PR DESCRIPTION
**Pull request by amihaiemil**
PR for #801 
Added the specified PMD rule and unit tests for it. 
Also, added todo puzzle to enhance the rule and also catch the `this.` case.

### CODE REVIEW

**Comment by davvd** [comment]
@amihaiemil Thanks, let me find someone who can review this pull request

**Comment by amihaiemil** [comment]
@krzyk its strange that davvd didnt assign a reviewer here yet. Is it blocked?

**Comment by krzyk** [comment]
@amihaiemil it usually takes at least a day or two for davvd to notice (and he doesn't work on weekends :)

**Comment by amihaiemil** [comment]
@krzyk usually a CR is assigned right when it says it's going to look for one... after 5 min or so. That's why I asked :D

**Comment by amihaiemil** [comment]
@davvd assign this PR for code review

**Comment by davvd** [comment]
> @davvd assign this PR for code review

@amihaiemil thanks

**Comment by krzyk** [comment]
@yegor256 do you know if we exceeded quota for the open source projects? I was almost 2 weeks but still no reviewer has been assigned here.

**Comment by krzyk** [comment]
@davvd please find a reviewer

**Comment by davvd** [comment]
> @davvd please find a reviewer

@krzyk OK, sure

**Comment by yegor256** [comment]
@krzyk yeah, we are over the quota. Let's wait a bit.

**Comment by krzyk** [comment]
@0crat in

**Comment by 0crat** [comment]
> @0crat in ([here](https://github.com/teamed/qulice/issues/811#issuecomment-385071828))

@krzyk Job [#811](https://github.com/teamed/qulice/issues/811) is now in scope, role is `REV`

File: qulice-pmd/src/main/java/com/qulice/pmd/rules/UseStringIsEmptyRule.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (c) 2011-2016, Qulice.com
```
Comment (krzyk) [inline qulice-pmd/src/main/java/com/qulice/pmd/rules/UseStringIsEmptyRule.java]: @amihaiemil You need to update it to 2018

File: qulice-pmd/src/test/java/com/qulice/pmd/UseStringIsEmptyRuleTest.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) 2011-2016, Qulice.com
```
Comment (krzyk) [inline qulice-pmd/src/test/java/com/qulice/pmd/UseStringIsEmptyRuleTest.java]: @amihaiemil Here also you need to update it to 2018

File: qulice-pmd/src/test/resources/com/qulice/pmd/StringLengthEqualsZero.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+public final class StringLengthEqualsZero {
+
+    private final String somestring;
+
+    public StringLengthEqualsZero(final String str) {
+        this.somestring = str;
+    }
+
+    public boolean sizeIsZero() {
+        return somestring.length == 0;
```
Comment (krzyk) [inline qulice-pmd/src/test/resources/com/qulice/pmd/StringLengthEqualsZero.java]: @amihaiemil should be `length()`

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `REV` are banned at this job; I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `REV` are banned at this job; I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by amihaiemil** [comment]
@krzyk FYI, changes done :)

**Comment by 0crat** [comment]
The job [#811](https://github.com/teamed/qulice/issues/811) is now out of scope

**Pull request merged**